ykcompile/src/lib.rs

```diff
@@ -43,6 +43,130 @@ lazy_static! {
     static ref PTR_SIZE: u64 = u64::try_from(mem::size_of::<usize>()).unwrap();
 }
 
+macro_rules! binop_as {
```

Comment:
You should be able to do all of this in one macro where you match over the operation name?

Reply:
I guess I could, but thought it might be a bit messy. I can try to see how it looks.

Reply:
I had a quick look and it will make that function messy and hard to read so I think I prefer them being split up.

Reply:
OK. Can we make the names more descriptive though? `as` and `md` confused me.

Reply:
How about `binop_add_sub` and `binop_mul_div`?

Reply:
sounds good.